roles/neutron-data-network/tasks/ipchanged.yml

```diff
@@ -19,6 +19,13 @@
 - name: configure neutron external services failover
   template: src=etc/ipchanged/add_floating_ip mode=0755
             dest=/etc/ipchanged/{{ ansible_default_ipv4.interface }}/{{ floating_ip }}/add
+  when: ansible_default_ipv4.interface != hostvars[inventory_hostname][primary_interface].device
+
+# Migrate Neutron services on internal floating IP failover when controllers are not gateway
+- name: configure neutron external services failover
+  template: src=etc/ipchanged/add_floating_ip mode=0755
+            dest=/etc/ipchanged/{{ ansible_default_ipv4.interface }}/{{ undercloud_floating_ip }}/add
+  when: ansible_default_ipv4.interface == hostvars[inventory_hostname][primary_interface].device
```

Comment:
You shouldn't need `hostvars[inventory_hostname]` those are default facts, but change `s/{{ floating_ip }}/{{ undercloud_floating_ip }}/` looks right.